gabbi/reporter.py

```diff
@@ -80,7 +80,8 @@ def addUnexpectedSuccess(self, test):
             self.stream.writeln('\t[unexpected success]')
 
     def getDescription(self, test):
-        name = test.test_data['name']
+        # Chop the test method (test_request) off the test.id().
+        name = test.id().rsplit('.', 1)[0]
```

Comment:
It's the reporter that is weird here, not test case, so I wanted to put the weirdness here.